internal/controller/backend_security_policy.go

```diff
@@ -2,39 +2,65 @@ package controller
 
 import (
 	"context"
+	"fmt"
+	"time"
 
+	egv1a1 "github.com/envoyproxy/gateway/api/v1alpha1"
 	"github.com/go-logr/logr"
 	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/kubernetes"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	aigv1a1 "github.com/envoyproxy/ai-gateway/api/v1alpha1"
+	backendauthrotators "github.com/envoyproxy/ai-gateway/internal/controller/rotators"
 )
 
 // backendSecurityPolicyController implements [reconcile.TypedReconciler] for [aigv1a1.BackendSecurityPolicy].
 //
 // This handles the BackendSecurityPolicy resource and sends it to the config sink so that it can modify configuration.
 type backendSecurityPolicyController struct {
-	client    client.Client
-	kube      kubernetes.Interface
-	logger    logr.Logger
-	eventChan chan ConfigSinkEvent
+	client       client.Client
+	kube         kubernetes.Interface
+	logger       logr.Logger
+	eventChan    chan ConfigSinkEvent
+	reconcileAll bool
 }
 
 func newBackendSecurityPolicyController(client client.Client, kube kubernetes.Interface, logger logr.Logger, ch chan ConfigSinkEvent) *backendSecurityPolicyController {
 	return &backendSecurityPolicyController{
-		client:    client,
-		kube:      kube,
-		logger:    logger,
-		eventChan: ch,
+		client:       client,
+		kube:         kube,
+		logger:       logger,
+		eventChan:    ch,
+		reconcileAll: true,
 	}
 }
 
+type patchBackendSecurityPolicy struct{}
+
 // Reconcile implements the [reconcile.TypedReconciler] for [aigv1a1.BackendSecurityPolicy].
-func (b backendSecurityPolicyController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (b *backendSecurityPolicyController) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.Result, err error) {
+	if b.reconcileAll {
+		var backendSecPolicyList aigv1a1.BackendSecurityPolicyList
+		err = b.client.List(ctx, &backendSecPolicyList)
+		if err != nil {
+			b.logger.Error(err, "failed to trigger refresh for existing backendSecPolicy resources")
+		} else {
+			for _, backendSecurityPolicy := range backendSecPolicyList.Items {
```

Comment:
This looks a bit odd to trigger the reconciliation for all security policies on a single resource reconciliation function. Controller runtime might have a way to resync all the resources periodically.

Reply:
> Controller runtime might have a way to resync all the resources periodically.

ha! if there's a way, then that should be much better

Reply:
@aabchoo sorry it seems like i was misunderstanding the default behavior

https://github.com/operator-framework/operator-sdk/issues/6035#issuecomment-1256360436


you can delete this all patching stuff!

Reply:
wait, this is operator-sdk... not sure about controller-runtime (operator-sdk using it under the hood?)

Reply:
yeah i think this applies to controller-runtime as well, let's delete this startup patching stuff